### PR TITLE
Upgrade dependencies, remove deprecated dependencies.

### PR DESCRIPTION
**Pull request by mashhurs**
## Description

Logstash has a `json` dependency and `json 1.8.6-java`. `json` dependency is originally coming from `aws-sdk-v1` where `logstash-mixin-aws` < 5.0.0 versions utilizes. This change:
 - upgrades `logstash-mixin-aws` to  `5.1.0`
 - removes unused `aws-sdk-v1` dependency.
 - upgrades `json` version to `2.6.3`
 - upgrades `fpm` to `1.14.1` where `13.x` versions.

Original PR: https://github.com/elastic/logstash/pull/12851

### CODE REVIEW

File: Gemfile.jruby-2.5.lock.release [hunk 1]
```diff
@@ -127,7 +124,7 @@ GEM
     filesize (0.2.0)
     fivemat (1.3.7)
     flores (0.0.8)
-    fpm (1.14.2)
+    fpm (1.15.0)
```
Comment (mashhurs) [inline Gemfile.jruby-2.5.lock.release:127]: This shouldn't be updated.
```suggestion
    fpm (1.14.2)
```

**Comment by mashhurs** [comment]
Failed CIs so far due to artifact miss, will re-run later.

**Comment by jsvd** [comment]
jenkins test this please

**Comment by mashhurs** [comment]
Failed CIs due to artifact miss.
`02:36:47 > could not find the current artifact from the artifact-api https://artifacts-api.elastic.co/v1/versions for 7.17.12`

**Review by jsvd — APPROVED** [review]
LGTM

**Pull request merged**